segmentation/include/pcl/segmentation/extract_clusters.h

```diff
@@ -47,19 +47,30 @@
 namespace pcl
 {
 
+namespace detail {
+template <typename PointT, typename Function>
+constexpr static bool is_functor_for_additional_filter_criteria_v =
+    pcl::is_invocable_r_v<bool,
+                          Function,
+                          const pcl::remove_cvref_t<pcl::PointCloud<PointT>>&,
```

Comment:
With `const` and reference collapsing rules, isn't this the same as `const remove_volatile_t<pcl::PointCloud<PointT>>&,`?

Reply:
It is. Which leads to the next question, why do you care if the user supplies a type which is volatile?

Reply:
I think it's more about `remove_cvref_t` being used to strip `const` and reference qualifiers, and nobody gives a thought about `volatile`. As far as I'm concerned I wouldn't consider `volatile` in such a case and would go with `const pcl::PointCloud<PointT>&?` directly.

Reply:
Just chiming in, being the author of the original snippet this is copied from: it's a mistake from when I was taking in the functor itself (`typename FunctorT`) and needed this to get arguments and constrain the input to `const PointCloud<PointT>&` (no volatile allowed). But the design changed, and got simplified thanks to the reviews but this managed to slip in.